### DISCUSSION TITLE
Migration Strategy

### DISCUSSION THREAD

**Discussion by doneill** [wix/react-native-ui-lib#2406] [General]
What is the strategy with migration, e.g. the TextField message to migrate.  Do we refactor to use the Incubator? Does the field ever come out of the Incubator? if it does, do we need to do another refactor?  I don't see deprecation notice and the message has been around for a long time, new to the libraries migration strategy so excuse my basic questions around this process, but any insight would be appreciated.

**Comment by doneill** [comment]
In addition: It looks like the new Incubator version of the TextField doesn’t have support for right icon anymore.  Is this planned to be supported?

**Reply by ethanshar** [reply]
Hi @doneill 
To make the migration easier, you can use the TextField (not from Incubator) and pass the `migrate` prop (it will internally use the Incubator version)

Then you should get warnings regarding the props that changed. Try following the instructions. 
For some props and behaviors I suggest following our docs. 
https://wix.github.io/react-native-ui-lib/docs/components/incubator/TextField

Specifically for rendering a right icon, you can now pass the `trailingAccessory` prop with whatever element you want. 

Good luck!

**Reply by JSantangelo-Octopus** [reply]
Hi! this link https://wix.github.io/react-native-ui-lib/docs/components/incubator/TextField is broken. Is there another way to get that docs?

**Reply by ethanshar** [reply]
The link was changed to https://wix.github.io/react-native-ui-lib/docs/components/form/TextField now that TextField is officially released (and out of Incubator) 
We also updated the link the migration docs
